doc/specs/#658 - WinGet Download.md

```diff
@@ -0,0 +1,209 @@
+
+Downloading the package's installer will still require that the package's installer hash be verified before becoming available to the user to interact with. The installer will be downloaded to the device with a `*.tmp` file extension that will be removed after the installer's hash has been validated. If the hash validation fails, the package's installer will be removed from the device.
+
+When downloading the package's installer, if a file with the same name exists the new download will overwrite the existing file.
+
+## UI/UX Design
+
+### WinGet Command Line
+Downloading an installer will output information relative to each step performed. Informing the user of any license agreements that must be accepted prior to download. Acceptance of license agreements will trigger the download to begin, displaying a progress bar that shows the download status. Upon download, the user will then be informed of the file hash validation status before being notified of the download status.
+
+The following is representative of the user experience.
+
+```PowerShell
+PS C:\> WinGet Download Microsoft.visualstudiocode
+Found Microsoft Visual Studio Code [Microsoft.VisualStudioCode] Version 1.73.1
+This application is licensed to you by its owner.
+Microsoft is not responsible for, nor does it grant any licenses to, third-party packages.
+Starting package download...
+   \
+Successfully verified installer hash
+Successfully downloaded
+
+PS: C:\> 
+```
+
+When the user runs the Help action on the WinGet Download command, the following information will be provided:
+
+```
+PS C:\> WinGet Download --help
+Windows Package Manager v1.1.1
+Copyright (c) Microsoft Corporation. All rights reserved.
+
+Downloads an identified package's installer, either found by searching a configured source or directly from a manifest. By default, the query must case-insensitively match the id, name, or moniker of the package. Other fields can be used by passing their appropriate option.
+
+The following command alias are available
+
+The following arguments are available:
+
+The following options are available
+    --id                            Filter results by id 
+    --name                          Filter results by name
+    --moniker                       Filter results by moniker
+    --scope                         Select install scope (user or machine)
+    --locale                        Locale to use (BCP47 format)
+    --filename                      The value to rename the downloaded executable file to
+    -v, --version                   Use the specified version; default is the latest version
+    -s, --source                    Find package using the specified source
```

Comment:
If we have `--source` we will also need `--accept-source-agreements` and the custom header for REST.

Reply:
Agree. Will add this to the spec.